python/trifinger_simulation/camera.py

```diff
@@ -11,9 +11,15 @@ class Camera(object):
     def __init__(
         self,
         camera_position,
-        camera_orientation,
+        camera_orientation=None,
         image_size=(270, 270),
         pybullet_client=pybullet,
+        field_of_view=52,
+        near_plane_distance=0.001,
+        far_plane_distance=100.0,
+        target_position=None,
+        camera_up_vector=None,
```

Comment:
If I understand this correctly, one needs to either give a `camera_orientation` or `target_position` and `camera_up_vector`?  This is not clear from the documentation and also error prone (e.g. what happens if I specify all of them or none?).

I think a better design would be to have separate factory methods, e.g. `Camera.createWithOrientation()` and `Camera.createWithTargetAndUpVector()`.

From what I saw, the rest of this PR is just porting fixes from rrc_simulation, which are also not directly related to this change on the Camera.  Would it be okay to move the additional changes in camera.py to a separate PR, so we can handle that separately and it does not block the other stuff?

Reply:
Okay, so I will revert the changes for the camera from here.

Reply:
Yes, I agree with you it is error-prone. I could add that check here though to address specifying all of them or none. Should I do that?

Reply:
I think separate methods would be better than checks that would only trigger at runtime.  In C++ this would be a typical use-case for overloads but this is unfortunately not possible in Python.

An alternative suggestion would be to wrap the camera pose in a separate structure.  Something like this:
```
class CameraPose
    def __init__(self, camera_position, target_position, camera_up_vector):
        self.camera_position = camera_position
        self.target_position = target_position
        self.camera_up_vector = camera_up_vector

    @classmethod
    def createWithTargetAndUpVector(cls, camera_position, target_position, camera_up_vector):
        return cls(camera_position, target_position, camera_up_vector)

    @classmethod
    def createWithQuaternion(cls, camera_position, camera_orientation):
        camera_rot = Rotation.from_quat(camera_orientation)
        target_position = camera_rot.apply([0, 0, 1])
        camera_up_vector = camera_rot.apply([0, -1, 0])
        return cls(camera_position, target_position, camera_up_vector)
```

Then pass this to `Camera` instead of the separate values.

Reply:
Done

Reply:
Okay, I will use this as a base for the CameraPose wrapper, thank you